cmd/crowdsec-cli/decisions.go

```diff
@@ -478,191 +481,259 @@ cscli decisions delete --type captcha
 	return cmdDecisionsDelete
 }
 
-func NewDecisionsImportCmd() *cobra.Command {
-	var (
+// decisionRaw is only used to unmarshall json/csv decisions
+type decisionRaw struct {
+	Duration string `csv:"duration,omitempty" json:"duration,omitempty"`
+	Origin   string `csv:"origin,omitempty" json:"origin,omitempty"`
+	Scenario string `csv:"reason,omitempty" json:"reason,omitempty"`
+	Scope    string `csv:"scope,omitempty" json:"scope,omitempty"`
+	Type     string `csv:"type,omitempty" json:"type,omitempty"`
+	Value    string `csv:"value" json:"value"`
+}
+
+func parseDecisionList(content []byte, format string) ([]decisionRaw, error) {
+	ret := []decisionRaw{}
+
+	switch format {
+	case "values":
+		log.Infof("Parsing values")
+		scanner := bufio.NewScanner(bytes.NewReader(content))
+		for scanner.Scan() {
+			value := strings.TrimSpace(scanner.Text())
+			ret = append(ret, decisionRaw{Value: value})
+		}
+		if err := scanner.Err(); err != nil {
+			return nil, fmt.Errorf("unable to parse values: '%s'", err)
+		}
+	case "json":
+		log.Infof("Parsing json")
+		if err := json.Unmarshal(content, &ret); err != nil {
+			return nil, err
+		}
+	case "csv":
+		log.Infof("Parsing csv")
+		if err := csvutil.Unmarshal(content, &ret); err != nil {
+			return nil, fmt.Errorf("unable to parse csv: '%s'", err)
+		}
+	default:
+		return nil, fmt.Errorf("invalid format '%s', expected one of 'json', 'csv', 'values'", format)
+	}
+	return ret, nil
+}
+
+
+func runDecisionsImport(cmd *cobra.Command, args []string) error  {
+	const (
 		defaultDuration = "4h"
 		defaultScope    = "ip"
 		defaultType     = "ban"
 		defaultReason   = "manual"
-		importDuration  string
-		importScope     string
-		importReason    string
-		importType      string
-		importFile      string
-		batchSize       int
 	)
 
-	var cmdDecisionImport = &cobra.Command{
+	flags := cmd.Flags()
+
+	input, err := flags.GetString("input")
+	if err != nil {
+		return err
+	}
+
+	importDuration, err := flags.GetString("duration")
+	if err != nil {
+		return err
+	}
+
+	importScope, err := flags.GetString("scope")
+	if err != nil {
+		return err
+	}
+
+	importReason, err := flags.GetString("reason")
+	if err != nil {
+		return err
+	}
+
+	importType, err := flags.GetString("type")
+	if err != nil {
+		return err
+	}
+
+	batchSize, err := flags.GetInt("batch")
+	if err != nil {
+		return err
+	}
+
+	format, err := flags.GetString("format")
+	if err != nil {
+		return err
+	}
+
+	var (
+		content []byte
+		fin	*os.File
+	)
+
+	// set format if the file has a json or csv extension
+	if format == "" {
+		if strings.HasSuffix(input, ".json") {
+			format = "json"
+		} else if strings.HasSuffix(input, ".csv") {
+			format = "csv"
+		}
+	}
+
+	if format == "" {
+		return fmt.Errorf("unable to guess format from file extension, please provide a format with --format flag")
```

Comment:
Could this line just sit at the bottom of the upper if statement

Reply:
No because it needs to check if it was detected. The alternative is not one if, but two nested ifs which is less readable.